picocrt/machine/arm/crt0.c

```diff
@@ -151,6 +151,19 @@ _start(void)
 #define _REASON(r) #r
 #define REASON(r) _REASON(r)
 
+static void arm_fault_write_reg(const char *prefix, unsigned reg)
```

Comment:
Would it make sense to place this in a private shared headset so that other architectures can also avoid printf?

Reply:
When another architecture wants to do that, it will be easy to make that change. No sense attempting to predict how this code will get used in the future..